test/DurableTask.SqlServer.Tests/Integration/UpgradeTests.cs

```diff
@@ -0,0 +1,306 @@
+            TimeSpan timeout = TimeSpan.FromSeconds(10);
+
+            // Test2, which was in Pending, should start and run to completion on its own.
+            instance = new OrchestrationInstance { InstanceId = "Test2" };
+            state = await client.WaitForOrchestrationAsync(instance, timeout);
+            Assert.Equal(OrchestrationStatus.Completed, state.OrchestrationStatus);
+            Assert.Equal(@"[""Hello, Tokyo!"",""Hello, London!"",""Hello, Seattle!""]", state.Output);
+
+            // Test4, which was Running, is resumed by the WaitForSignal orchestrations
+            string data = Guid.NewGuid().ToString();
+            instance = new OrchestrationInstance { InstanceId = "Test4" };
+            await client.RaiseEventAsync(instance, "signal", data);
+            state = await client.WaitForOrchestrationAsync(instance, TimeSpan.FromSeconds(10));
+            Assert.Equal(OrchestrationStatus.Completed, state.OrchestrationStatus);
+            Assert.Equal($"\"{data}\"", state.Output);
+
+            // Test5 is waiting on Test5-child, which is waiting for an external event.
+            data = Guid.NewGuid().ToString();
+            instance = new OrchestrationInstance { InstanceId = "Test5-child" };
+            await client.RaiseEventAsync(instance, "signal", data);
+            state = await client.WaitForOrchestrationAsync(instance, TimeSpan.FromSeconds(10));
+            Assert.Equal(OrchestrationStatus.Completed, state.OrchestrationStatus);
+            Assert.Equal($"\"{data}\"", state.Output);
+            instance = new OrchestrationInstance { InstanceId = "Test5" };
+            state = await client.WaitForOrchestrationAsync(instance, TimeSpan.FromSeconds(10));
+            Assert.Equal(OrchestrationStatus.Completed, state.OrchestrationStatus);
+            Assert.Equal($"\"{data}\"", state.Output);
+
+            // Test6, which was cut off during an activity execution, should run to completion on its own.
+            OrchestrationState test6 = await client.WaitForOrchestrationAsync(
+                new OrchestrationInstance { InstanceId = "Test6" },
+                TimeSpan.FromSeconds(10));
+            Assert.Equal(OrchestrationStatus.Completed, test6.OrchestrationStatus);
+            Assert.Equal(@"[""Hello, Tokyo!"",""Hello, London!"",""Hello, Seattle!""]", test6.Output);
+        }
+
+        class SimpleObjectCreator<T> : ObjectCreator<T>
+        {
+            readonly Type type;
+
+            public SimpleObjectCreator(string name, Type type)
+            {
+                this.Name = name;
+                this.type = type;
+            }
+
+            public override T Create()
+            {
+                return (T)Activator.CreateInstance(this.type)!;
+            }
+        }
+
+        // NOTE: Must be kept consistent with the version defined in tools/TestDBGenerator/Orchestrations.cs
+        class ActivitySequence : TaskOrchestration<string[], object>
```

Comment:
if these need to be consistent, is it possible for us to have a single definition that is referenced across both files?

Reply:
My reasoning for keeping separate copies of these is that one copy must never be changed (it needs to represent a snapshot of the type of code we supported in the v1.0.0 release of the SQL provider) whereas the other can be changed as appropriate based on future product updates we make.

Reply:
I see. So essentially the one that should not change is `Orchestrations.cs` and the one that may be changed is the one in the tests?

Reply:
Right - it's fine to change the test code (provided it's semantically equivalent).

Reply:
In any case, could we document as a comment in code which file is meant to remain static? That wasn't immediately obvious to me and it seems calling out. @cgillum

Reply:
I think this is my only blocking suggestion at this point.

Reply:
I've added comments at the top of TestDBGenerator/Orchestrations.cs, which should hopefully help clarify this.